### PR TITLE
Issue #13809: Kill mutation in CommonUtil

### PR DESCRIPTION
**Pull request by Kevin222004**
Issue #13809: Kill mutation in CommonUtil

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/utils/CommonUtil.java [hunk 1]
```diff
@@ -123,20 +123,8 @@ public static boolean matchesFileExtension(File file, String... fileExtensions)
...
-                final String extension = fileExtensions[i];
-                if (extension.startsWith(EXTENSION_SEPARATOR)) {
-                    withDotExtensions[i] = extension;
-                }
-                else {
-                    withDotExtensions[i] = EXTENSION_SEPARATOR + extension;
-                }
-            }
-
             final String fileName = file.getName();
```
Comment (Kevin222004) [inline src/main/java/com/puppycrawl/tools/checkstyle/utils/CommonUtil.java:126]: @romani do we really required to do this at the end `fileName.endsWith(fileExtension)` we are checking only wether file names end with fileExtension.

Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/utils/CommonUtil.java:126]: I am surprised on this code 
The only place where we do mentioning `.` in extension name is at https://checkstyle.org/config.html#TreeWalker_Properties

All other places are without dot.

Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/utils/CommonUtil.java:126]: Extension is not a standard , it is just agreement that file name ends with some characters.

We should not parse java file that is named as `Test,java` so looks like we missed test on names without dots as separator of name and extension.

Comment (Kevin222004) [reply src/main/java/com/puppycrawl/tools/checkstyle/utils/CommonUtil.java]: Test added

**Review by romani — APPROVED** [review]
Ok to merge

**Pull request merged**